utils.py

```diff
@@ -123,3 +131,24 @@ def write_results(self, results):
 
     def close(self):
         self.csv_file.close()
+
+
+class BestModelRecords:
+
+    def __init__(self, metrics_list):
+        metrics = [ metric.strip() for metric in metrics_list.split(',') ]
+        self.record_values = { metric_name : None for metric_name in metrics }
+
+    def better_than_previous(self, metric_name, metric_value):
+        record_value = self.record_values[metric_name]
+        if record_value is None:
+            return True
+        return metric_value > record_value if METRICS_MAXIMIZE[metric_name] else metric_value < record_value
```

Comment:
Why do we use `METRICS_MAXIMIZE` here? What is the relation of this variable to `metrics_list` which (also?) stores metrics to optimize for?

Reply:
I see your point. The reason why `METRICS_MAXIMIZE` is a dict instead of a list is that we need `METRICS_MAXIMIZE.keys()` in args help.